### PR TITLE
Connect: Table/field selection

### PR DESCRIPTION
**Pull request by erinkcochran87**
This PR adds documentation to the Connect API for:

- Endpoints for streams (table) and selecting fields in streams
- An endpoint for source connection checks
- How-tos for selecting streams and fields
- Adds a 'tutorial' layout and on-page sidebar
- Adds an on-page sidebar to 'general' layouts

### CODE REVIEW

**Comment by erinkcochran87** [comment]
**Previews**: 
- [Field compatibility and selection rules](https://deploy-preview-232--stitchdocs.netlify.com/docs/stitch-connect/guides/field-selection-compatibility-rules)
- [Select streams and fields with the Connect API](https://deploy-preview-232--stitchdocs.netlify.com/docs/stitch-connect/guides/select-tables-and-fields-with-connect-api)

File: _connect-content/guides/api/select-tables-and-fields.md [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+# -------------------------- #
+#         GUIDE INTRO        #
+# -------------------------- #
+
+requirements:
+  - item: |
+      **Valid credentials**. Using the Connect API requires a valid [access token]({{ site.data.connect.api.section | flatify | prepend: site.baseurl | append: site.data.connect.api.access-api }}), which is obtained through partner credentials.
+  - item: |
+      **Access to a source**. This guide will use a [Shopify SaaS source]({{ site.data.connect.api.section | flatify | prepend: site.baseurl | append: site.data.connect.data-structures.source-form-properties.section |  append: "-shopify-object" }}) as an example, but any source type will work. Refer to the [Source Form Properties documentation]({{ site.data.connect.api.section | flatify | prepend: site.baseurl | append: site.data.connect.data-structures.source-form-properties.section }}) for the sources supported by Connect.
+
```
Comment (iterati) [inline _connect-content/guides/api/select-tables-and-fields.md]: This actually only works on a source with a field_selection step. Might want to include that info here.

File: _connect-content/guides/api/select-tables-and-fields.md [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+          curl -X GET {{ site.data.connect.api.base-url | strip_newlines }}{{ site.data.connect.core-objects.sources.retrieve.name | flatify | replace: "{source_id",source-id | remove: right-bracket | strip_newlines }}
+               -H 'Content-Type: application/json' \
+               -H 'Authorization: Bearer <API_TOKEN>'
+          ```
+
+          The response will be the source's [`report_card` object]({{ site.data.connect.api.section | flatify | prepend: site.baseurl | append: site.data.connect.data-structures.report-cards.source.section }}). In this example, the `current_step` is `4`, which corresponds to the `field_selection` step:
+
+          ```json
+          {{ site.data.connect.code-examples.source-report-cards.shopify | replace: "<STEP_NUMBER>","4" }}
+          ```
```
Comment (iterati) [inline _connect-content/guides/api/select-tables-and-fields.md]: fully_configured sources can also have selections altered. And (currently) in all cases field_selection and fully_configured are the last two steps when a source support field selections.

File: _connect-content/guides/api/select-tables-and-fields.md (hunk 1 above)
Comment (erinkcochran87) [reply _connect-content/guides/api/select-tables-and-fields.md]: Good point! I'll add that in.

File: _connect-content/guides/api/select-tables-and-fields.md (hunk 2 above)
Comment (erinkcochran87) [reply _connect-content/guides/api/select-tables-and-fields.md]: Also a good point! I'll add that in, too.

**Pull request merged**